oak_runtime/src/introspect.rs

```diff
@@ -141,6 +177,21 @@ fn handle_request(
             "application/x-protobuf".parse().unwrap(),
         );
 
+        return Ok(response);
+    // Serve the webpack `dist` directory containing client files
+    } else if let Some((mut file, content_type)) = find_client_file(path) {
+        let mut data = Vec::new();
+        if file.read_to_end(&mut data).is_err() {
```

Comment:
It feels slightly odd that most of the work and error checks live in `find_client_file`, except for the check that we can read the file contents.  

Maybe subsume this check into `find_client_file` (which then returns `Option<(Vec<u8>, String)>`)?  (A file read failure would then just trigger an `err!` rather than a 500 return, though.)  WDYT?

Reply:
Fair. The reason I wrote it that way is to preserve accuracy the HTTP status message.

Were I to move the check up into `find_client_file`, a failure while reading a file would result in a `404` status code. Which technically wouldn't be accurate, since the file does exist, there was just an error while reading it. 

In terms of responsibility, you can think of `find_client_file` as handling "is there a file we want to serve", and `handle_request` of actually assembling a response based on that. 

An alternative I thought about was to make `find_client_file` itself return `Option<Response<Body>>`, but I didn't really like that either.

Reply:
That said, by moving bundling assets into the binary, as suggested @tiziano88 we move the concern of failure while reading files to compile time :)

Reply:
Resolved with https://github.com/project-oak/oak/pull/1382/commits/2903b75180fae7d72ba9816b4154220f1a55780a